### PR TITLE
Allow custom headers to be set in Server

### PR DESCRIPTION
**Pull request by RohitK89**
This change allows users to pass in custom headers when instantiating a Server object, without needing to create an entire client by themselves.

This feature allows users to leverage the well-formed default client that has retry and backoff logic baked in already in situations where they need to pass in extra headers for their own tracking, or more critically, for authenticating against their Horizon provider.

# Testing

I tested this change locally and validated that tests are passing.

```
================================================================================================================= 972 passed, 62 skipped, 82 warnings in 6.23s =================================================================================================================
```

### CODE REVIEW

**Comment by RohitK89** [comment]
CC @overcat

File: stellar_sdk/client/requests_client.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ def __init__(
         backoff_factor: float = DEFAULT_BACKOFF_FACTOR,
         session: Session = None,
         stream_session: Session = None,
+        custom_headers: Dict[str, str] = {},
```
Comment (overcat) [inline stellar_sdk/client/requests_client.py:53]: I personally tend to modify it as follows, although this probably won't cause a bug in the current situation, for more information: https://stackoverflow.com/a/26320938

```suggestion
        custom_headers: Dict[str, str] = None,
```

File: stellar_sdk/server.py [hunk 2]
```diff
@@ -39,17 +39,20 @@ class Server(BaseServer):
         (ex. ``"https://horizon-testnet.stellar.org"`` for test network,
         ``"https://horizon-testnet.stellar.org"`` for public network)
     :param client: Http client used to send the request
+    :param custom_request_headers: Any headers that should be added when making HTTP requests to the
```
Comment (overcat) [inline stellar_sdk/server.py:42]: Personally, I prefer not to modify this file. Modifying headers is not a common requirement, and not every client implements the function of modifying headers. If a user has such a need, then he should create a client instance himself, it's not difficult in my opinion.

**Review by overcat — CHANGES_REQUESTED** [review]

**Comment by overcat** [comment]
@RohitK89 Sorry, I closed this PR by mistake, can you re-push the code and reopen this PR? 🤦

**Pull request closed without merging**